sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/models/ProgressReceiver.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callingserver.models;
+
+import reactor.core.publisher.Flux;
+
+/**
+ * A {@code ProgressReceiver} is an object that can be used to report progress on network transfers. When specified on
+ * transfer operations, the {@code reportProgress} method will be called periodically with the total number of bytes
+ * transferred. The user may configure this method to report progress in whatever format desired. It is recommended
+ * that this type be used in conjunction with {@link ProgressReporter#addProgressReporting(Flux, ProgressReceiver)} to
+ * enable reporting on sequential transfers. Note that any method accepting a {@link ParallelDownloadOptions} will use
+ * the {@code ProgressReceiver} specified there and will handle the logic to coordinate the reporting between parallel
+ * operations.
+ */
+public interface ProgressReceiver {
```

Comment:
Comment from @alzimmermsft in [this PR](https://github.com/Azure/azure-sdk-for-java-pr/pull/1324):
_Do you believe this functionality would be worthwhile existing in azure-core? I know this is copied from Storage but maybe it would be better condensing it._

Reply:
It probably needs to be accessible for more resources than Storage and us. So it may be worth the discussion to do that.
Anyway, I'd like to track such change in a different PR, given the impact with Storage and CallingServer.

Reply:
Agreed, this doesn't need to be done eagerly but I'd like to close on the discussion before this needs to GA